tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx

```diff
@@ -1340,15 +1341,14 @@ public:
 
    void Finalize()
    {
-      if (fOutputFile && fOutputTree) {
-         // because TTree::Write writes in gDirectory, not in fDirectory
-         fOutputTree->AutoSave("flushbaskets");
-         // must destroy the TTree first, otherwise TFile will delete it too leading to a double delete
-         fOutputTree.reset();
-         fOutputFile->Close();
-      } else {
-         Warning("Snapshot", "A lazy Snapshot action was booked but never triggered.");
-      }
+      R__ASSERT(fOutputTree != nullptr);
+      R__ASSERT(fOutputFile != nullptr);
```

Comment:
Is `R__ASSERT` also valid for non-debug code? I.e. is it always evaluated?

Reply:
I hope not! :) These are debug asserts, if `Finalize` is called we really expect `fOutputTree` and `fOutputFile` to not be null (they are set in `Initialize`, a few lines above).

Reply:
OK, thanks for the clarification :-)